crates/polars-ops/src/chunked_array/list/namespace.rs

```diff
@@ -513,6 +514,134 @@ pub trait ListNameSpaceImpl: AsList {
         list_ca.apply_amortized(|s| s.as_ref().drop_nulls())
     }
 
+    #[cfg(feature = "list_pad")]
+    fn lst_pad_start(&self, fill_value: &Column, length: &Column) -> PolarsResult<ListChunked> {
+        let ca = self.as_list();
+        let inner_dtype = ca.inner_dtype();
+        let fill_dtype = fill_value.dtype();
+        let super_type = try_get_supertype(inner_dtype, fill_dtype)?;
+
+        let dtype = &DataType::List(Box::new(super_type.clone()));
+        let ca = ca.cast(dtype)?;
+        let ca = ca.list().unwrap();
+
+        let length = if length.len() == 1 {
+            &length.new_from_index(0, ca.len())
+        } else {
+            length
+        };
+        let length = length.strict_cast(&DataType::UInt64)?;
+        let mut length = length.u64()?.into_iter();
+
+        let fill_value = if fill_value.len() == 1 {
+            &fill_value.new_from_index(0, ca.len())
+        } else {
+            fill_value
+        };
+        let fill_value = fill_value.cast(&super_type)?;
+
+        let out: ListChunked = match super_type {
```

Comment:
This entire `match` should probably be reduced by calling one function per datatype, which will be also used for `list.pad_end()` (in another PR).